File: assets/examples/js/film_db_rag.mjs
// Node 18+ is required (Node 20 is preinstalled on dev-host via NVM)
const SCHEMA = process.env.SCHEMA_NAME || 'filmdata1'
const PROXY_HOST = (process.env.PROXY_HOST || '').trim()

function baseUrl() {
  if (PROXY_HOST) {
    if (PROXY_HOST.startsWith('http://') || PROXY_HOST.startsWith('https://')) return PROXY_HOST
    return `https://${PROXY_HOST}`
  }
  return 'http://postgrest:3000'
}

async function getJson(url) {
  const res = await fetch(url, { headers: { 'Accept-Profile': SCHEMA } })
  if (!res.ok) throw new Error(`HTTP ${res.status}`)
  return await res.json()
}

async function main() {
  const base = baseUrl()
  console.log(`Using schema=${SCHEMA}`)
  console.log(`Proxy base=${base}`)
  const url = `${base}/film?select=film_id,title,description&order=title.asc&limit=5`
  const rows = await getJson(url)
  console.log(JSON.stringify(rows, null, 2).slice(0, 2000))
  console.log('\nOK')
}

main().catch((e) => { console.error('ERROR:', e); process.exit(2) })

File: assets/examples/python/film_db_rag.py
#!/usr/bin/env python3
import os
import sys
import json
from urllib.parse import urljoin

try:
    from dotenv import load_dotenv
    load_dotenv()
except Exception:
    pass

SCHEMA = os.getenv("SCHEMA_NAME", "filmdata1")
PROXY_HOST = (os.getenv("PROXY_HOST") or "").strip()

def base_url():
    if PROXY_HOST:
        scheme = "https" if "://" not in PROXY_HOST else PROXY_HOST.split("://")[0]
        host = PROXY_HOST.split("://")[-1]
        return f"{scheme}://{host}"
    # in-cluster fallback
    return "http://postgrest:3000"

def get(url: str):
    import requests
    h = {"Accept-Profile": SCHEMA}
    r = requests.get(url, headers=h, timeout=20)
    r.raise_for_status()
    return r.json()

def list_films(limit=5):
    url = base_url() + f"/film?select=film_id,title,description&order=title.asc&limit={limit}"
    return get(url)

def main():
    print(f"Using schema={SCHEMA}")
    print(f"Proxy base={base_url()}")
    try:
        rows = list_films(5)
        print(json.dumps(rows, indent=2)[:2000])
        print("\nOK: fetched films")
    except Exception as e:
        print(f"ERROR: {e}")
        sys.exit(2)

if __name__ == "__main__":
    main()

